Allow configuring redirect handler using the @ClientRedirectHandler annotation

Sgitario opened the pull request:
Follow-up task of https://github.com/quarkusio/quarkus/pull/31142#issuecomment-1429447206.

Sgitario commented:
@geoand I guess these changes are what you had in mind in your comment https://github.com/quarkusio/quarkus/pull/31142#issuecomment-1429447206, right?

geoand commented on extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java:
```diff
@@ -557,6 +492,108 @@ void addRestClientBeans(Capabilities capabilities,
         }
     }
 
+    private void populateClientExceptionMappersFromAnnotations(BuildProducer<GeneratedClassBuildItem> generatedClasses,
```
The method name probably needs to be updated :)

Sgitario replied:
I've renamed it to `populateClientExceptionMapperFromAnnotations`. 
We have two methods:
- `populateClientExceptionMapperFromAnnotations` to populate `@ClientExceptionMapper`
- `populateClientRedirectHandlerFromAnnotations` to populate `@ClientRedirectHandler`

geoand reviewed (commented):
Very nice!

I just added one small comment

Sgitario committed:
Allow configuring redirect handler using the @ClientRedirectHandler ann

Follow-up task of https://github.com/quarkusio/quarkus/pull/31142

Related comment https://github.com/quarkusio/quarkus/pull/31142#issuecomment-1429447206

geoand reviewed (approved):
💪🏼
